File: CPP/L3/Circle.cpp
#include "Circle.h"

Circle::Circle(double r): radius(r) {};
Circle::Circle(char* r){
	radius = atof(r);
	if (radius <= 0) throw "Circle radius must have a positive value";
}

double Circle::getArea() const {
	return radius*radius*PI;
}
double Circle::getPerimeter() const {
	return 2*radius*PI;
}
void Circle::print(){
	std::cout << "Circle Area: " << getArea() << " Perimeter: " << getPerimeter() << "\n";
}

File: JAVA/L3/circle.java
public class circle extends shape{
	double radius;

	circle(){
		radius = 0;
	}
	circle(double radius){
		this.radius = radius;
	}

	public double getArea(){
		return Math.PI * radius * radius;
	}
	public double getPerimeter(){
		return 2 * Math.PI * radius;
	}
	@Override
	public String toString(){
		return "circle with radius: " + radius;
	}
}